tests/BigIntTests/BigIntTests.swift

```diff
@@ -124,14 +124,21 @@ class BigIntTests: XCTestCase {
         compare(-7, -4, r: 1, op: /)
     }
 
-    func testModulo() {
+    func testRemainder() {
         compare(0, 1, r: 0, op: %)
         compare(0, -1, r: 0, op: %)
         compare(7, 4, r: 3, op: %)
         compare(7, -4, r: 3, op: %)
         compare(-7, 4, r: -3, op: %)
         compare(-7, -4, r:-3, op: %)
     }
+  
+    func testModulo() {
+        XCTAssertEqual(BigInt.modulus(a: BigInt(abs: 22, negative: false), b: BigInt(abs: 5, negative: false)), BigInt(abs: 2, negative: false))
+        XCTAssertEqual(BigInt.modulus(a: BigInt(abs: 22, negative: true), b: BigInt(abs: 5, negative: false)), BigInt(abs: 3, negative: false))
+        XCTAssertEqual(BigInt.modulus(a: BigInt(abs: 22, negative: false), b: BigInt(abs: 5, negative: true)), BigInt(abs: 3, negative: true))
+        XCTAssertEqual(BigInt.modulus(a: BigInt(abs: 22, negative: true), b: BigInt(abs: 5, negative: true)), BigInt(abs: 2, negative: true))
```

Comment:
`compare` should help make this a little more readable; alternatively, you could also use integer literals here — `BigInt.modulus` nicely constraints the expression, so type inference deduces the `BigInt`s:

```swift
XCTAssertEqual(BigInt.modulus(-22, 5), 3)
```

(The expected values here match those of a division operator that rounds towards -infinity, like `fdiv` in GMP. This is certainly a valid operation, but `modulus` may not be the best name for it. Are you sure this is the semantics you want?)